pkg/adapter/v2/test/test_client.go

```diff
@@ -62,11 +71,21 @@ func (c *TestCloudEventsClient) Send(ctx context.Context, out event.Event) proto
 	c.lock.Lock()
 	defer c.lock.Unlock()
 	// TODO: improve later.
+	bytes, _ := json.Marshal(out)
+	if err := json.Unmarshal(bytes, &eventData); err != nil {
+		fmt.Printf("json unmarshal error %s:", err)
+	}
 	c.sent = append(c.sent, out)
-	if len(c.resultSend) != 0 {
-		resp := c.resultSend[0]
-		c.resultSend = c.resultSend[1:]
-		return resp
+	if eventData.Type == "unit.type" {
+		return http.NewResult(200, "%w", protocol.ResultACK)
+	} else if eventData.Type == "unit.retries" {
+		var attempts []protocol.Result
+		attempts = append(attempts, http.NewResult(500, "%w", protocol.ResultACK))
```

Comment:
Are these here reversed and L109-110? In other cases 200 => ResultACK and errors are NACK? If this is intentional for some reason, perhaps add a comment here explaining why?
As in, should this be ResultNACK and L84 ResultACK?

Reply:
@vaikas I just took the result values from the `cloudevent_test.go` file and put it here in `test_client` file for the conditions because I wanted correct Result values for my conditions. Since I added the various results for various type of tests, I removed that from the `cloudevents_test.go` file. I haven't added anything new but just shuffled it around to other file based on changes I have made.

Reply:
Sorry, what I meant is, on Line 80, 90 and 106

We do this:
http.NewResult(200, "%w", protocol.ResultACK)

But here you use ResultNACK yet return 200:
http.NewResult(200, "%w", protocol.ResultNACK)
Or, put another way, why is this is not also:
http.NewResult(200, "%w", protocol.ResultACK)

Reply:
I think you're right.. I should have put `http.NewResult(200, "%w", protocol.ResultACK` but I had put the report retry event count code in the `..else` condition of this check `cloudevents.IsACK(result)`. I think that was wrong, it should be able to report the retry events regardless whether the result isACK or not. So I  moved it outside and changed the test to send `protocol.ResultACK`